src/api/index.ts

```diff
@@ -31,17 +39,42 @@ export class ArgoCDApiClient implements ArgoCDApi {
     return `${proxyUrl}${this.proxyPath}`;
   }
 
+  private async fetchDecode<A, O, I>(url: string, typeCodec: t.Type<A, O, I>) {
```

Comment:
I'm, using io-ts library which is very good, but can introduce a little bit of traction for maintenance for people that are not familiar with it.

I'm also not sure it this `fetchDecode` method should be copied and pasted in each project that is using it, or extracted to shared library. I'm not sure if it justifies adding extra complexity and indirection.

Reply:
Maybe a few lines in Readme that this is used to validate API reqs would be good. I think it's a good approach regardless